components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests

```diff
@@ -88,6 +87,7 @@ function cleanup() {
   fi
 
   if [[ -z "${EKS_EXISTING_CLUSTER}" ]]; then
+    delete_generated_role
```

Comment:
script should delete resources it created and hence previous logic was correct.

If role already exists for an existing cluster and we want to avoid deleting it, I think we should have an environment variable so that its not created as well

maybe we could avoid creating role and setting up odic for existing clusters

Reply:
yep